src/drv_pwm.c

```diff
@@ -187,19 +189,19 @@ static void pwmOCConfig(TIM_TypeDef *tim, uint8_t channel, uint16_t value)
     switch (channel) {
         case TIM_Channel_1:
             TIM_OC1Init(tim, &TIM_OCInitStructure);
-            TIM_OC1PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC1PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_2:
             TIM_OC2Init(tim, &TIM_OCInitStructure);
-            TIM_OC2PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC2PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_3:
             TIM_OC3Init(tim, &TIM_OCInitStructure);
-            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_4:
             TIM_OC4Init(tim, &TIM_OCInitStructure);
-            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Disable);

```

Comment:
uh, and what happens when this useless pwm is NOT enabled?
On Mar 11, 2015 10:11 PM, "luggi" <notifications@github.com> wrote:

> In src/drv_pwm.c
> <https://github.com/multiwii/baseflight/pull/256#discussion_r26210002>:
>
> >              break;
> >          case TIM_Channel_4:
> >              TIM_OC4Init(tim, &TIM_OCInitStructure);
> > -            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Enable);
> > +            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Disable);
>
> this to disable the ouput compare register buffering, it only gets updated
> on a timer overflow i think, which doesn't happen with syncpwm disabled
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/multiwii/baseflight/pull/256/files#r26210002>.
>